resources/assets/lib/utils/lang.tsx

```diff
@@ -2,12 +2,33 @@
 // See the LICENCE file in the repository root for full licence text.
 
 import Lang from 'lang.js';
+import * as React from 'react';
 import { formatNumber } from 'utils/html';
 import { present } from 'utils/string';
 
 type Replacement = string | number;
 type Replacements = Partial<Record<string, Replacement>>;
 
+export function joinComponents(array: React.ReactNode[], key = 'common.array_and') {
+  switch (array.length) {
+    case 0:
+      return '';
+    case 1:
+      return array[0];
+    case 2:
+      return <>{array[0]}{trans(`${key}.two_words_connector`)}{array[1]}</>;
+  }
+
+  const nodes = [array[0]];
+  const lastIndex = array.length - 1;
+  for (let i = 1; i < lastIndex - 1; i++) {
+    nodes.push(<React.Fragment key={i}>{trans(`${key}.words_connector`)} {array[i]}</React.Fragment>);
```

Comment:
this will cause key conflict if the first element (which has key specified by the caller) has key value of 1. Similarly for the last item below. It's probably better to just explicitly set all the keys in this function itself instead. 

Also may want to wrap the return with key-less fragment so the caller doesn't need to make sure adjacent components have the expected keys.